includes/class-classic-conversion.php

```diff
@@ -0,0 +1,650 @@
+<?php
+/**
+ * Core setup, site hooks and filters.
+ *
+ * @package Go
+ */
+
+namespace Go;
+
+use WP_Theme_JSON;
+
+/**
+ * This class handles the conversion of the classic theme to the block based theme.
+ */
+class Classic_Conversion {
+	/**
+	 * The current globally available instance (if any).
+	 *
+	 * @var static
+	 */
+	protected static $instance;
+
+	/**
+	 * The base path for the theme.
+	 *
+	 * @var string
+	 */
+	protected $base_path;
+
+	/**
+	 * The new navigation locations.
+	 *
+	 * @var array
+	 */
+	protected $nav_menu_locations;
+
+	/**
+	 * Get the globally available instance.
+	 *
+	 * @return static
+	 */
+	public static function get_instance() {
+		if ( \is_null( static::$instance ) ) {
+			static::$instance = new static();
+		}
+		return static::$instance;
+	}
+
+	/**
+	 * Register the path bindings based on the main functions file.
+	 *
+	 * @param string $functions_file The full path to the main functions file.
+	 */
+	public function set_base_path( $functions_file ) {
+		$this->base_path = trailingslashit( dirname( $functions_file ) );
+	}
+
+
+	/**
+	 * Get the base path of the theme.
+	 *
+	 * @param string $path path from the root.
+	 *
+	 * @return string
+	 */
+	public function base_path( $path = '' ) {
+		return rtrim( $this->base_path, DIRECTORY_SEPARATOR ) . ( '' !== $path ? DIRECTORY_SEPARATOR . ltrim( $path, DIRECTORY_SEPARATOR ) : '' );
+	}
+
+	/**
+	 * Determines if a post exists based on name and type.
+	 *
+	 * @global wpdb $wpdb WordPress database abstraction object.
+	 *
+	 * @param string $name   Post name.
+	 * @param string $type    Optional. Post type.
+	 *
+	 * @return int Post ID if post exists, 0 otherwise.
+	 */
+	public function post_exists( $name, $type ) {
+		global $wpdb;
+
+		$post_name = wp_unslash( sanitize_post_field( 'post_name', $name, 0, 'db' ) );
+		$post_type = wp_unslash( sanitize_post_field( 'post_type', $type, 0, 'db' ) );
+
+		$query = "SELECT ID FROM $wpdb->posts WHERE 1=1";
+		$args  = array();
+
+		$query .= ' AND post_name = %s';
+		$args[] = $post_name;
+
+		$query .= ' AND post_type = %s';
+		$args[] = $post_type;
+
+		return (int) $wpdb->get_var( $wpdb->prepare( $query, $args ) );
```

Comment:
The code is correctly building out a query with placeholders, and utilzing wpdb->prepare function as the WP Coding Standards require. Unfortunately it seems like this way of building the query is confusing the linter. Maybe we should re-write this so the query lives inside of `prepare` this satisfies the liter and is still quite readable. 

```suggestion
				return (int) $wpdb->get_var(
			$wpdb->prepare(
				"SELECT ID FROM $wpdb->posts WHERE 1=1  AND post_name = %s AND post_type = %s",
				array( $post_name, $post_type ),
			)
		);
```

Reply:
I'm going to phpcs:ignore this one as well. The snippet was pulled directly from core and is properly prepared. I've also added a link in the phpdoc for reference.